src/igraph/__init__.py

```diff
@@ -382,30 +382,35 @@ def delete_edges(self, *args, **kwds):
         """Deletes some edges from the graph.
 
         The set of edges to be deleted is determined by the positional and
-        keyword arguments. If any keyword argument is present, or the
-        first positional argument is callable, an edge
-        sequence is derived by calling L{EdgeSeq.select} with the same
-        positional and keyword arguments. Edges in the derived edge sequence
-        will be removed. Otherwise the first positional argument is considered
-        as follows:
+        keyword arguments. If the functions is called without any arguments,
+        all edges are deleted. If any keyword argument is present, or the
+        first positional argument is callable, an edge sequence is derived by
+        calling L{EdgeSeq.select} with the same positional and keyword
+        arguments. Edges in the derived edge sequence will be removed.
+        Otherwise the first positional argument is considered as follows:
 
           - C{None} - deletes all edges
+            @deprecated: replaced by no arguments since igraph 0.8.3
           - a single integer - deletes the edge with the given ID
           - a list of integers - deletes the edges denoted by the given IDs
           - a list of 2-tuples - deletes the edges denoted by the given
             source-target vertex pairs. When multiple edges are present
             between a given source-target vertex pair, only one is removed.
         """
         if len(args) == 0 and len(kwds) == 0:
-            raise ValueError("expected at least one argument")
-        if len(kwds)>0 or (hasattr(args[0], "__call__") and \
+            # This means delete all
+            edge_seq = None
+        elif len(kwds) > 0 or (hasattr(args[0], "__call__") and \
```

Comment:
Now that we are here, can we replace `hasattr(args[0], "__call__")` with `callable(args[0])`? Python did not have `callable()` when I wrote it many years ago, but now it does.

Reply:
yes